ouroboros-consensus-cardano/tools/db-analyser/Analysis.hs

```diff
@@ -0,0 +1,215 @@
+
+instance TPraosCrypto c => HasAnalysis (ShelleyBlock c) where
+    countTxOutputs blk = case Shelley.shelleyBlockRaw blk of
+      SL.Block _ (SL.TxSeq txs) -> sum $ fmap countOutputs txs
+    blockHeaderSize =
+      fromIntegral . SL.bHeaderSize . SL.bheader . Shelley.shelleyBlockRaw
+    blockTxSizes blk = case Shelley.shelleyBlockRaw blk of
+      SL.Block _ (SL.TxSeq txs) ->
+        toList $ fmap (fromIntegral . BL.length . SL.txFullBytes) txs
+    mkTopLevelConfig [configFile] _onMainNet = do
+      genesis  <- either (error . show) return =<< Aeson.eitherDecodeFileStrict' configFile
+      return $ mkShelleyTopLevelConfig genesis
+    mkTopLevelConfig ls _onMainNet =
+      error $
+        "A single genesis file is needed for pure Shelley. Given " ++ show ls
+
+mkShelleyTopLevelConfig :: forall c. TPraosCrypto c
+                        => ShelleyGenesis c
+                        -> TopLevelConfig (ShelleyBlock c)
+mkShelleyTopLevelConfig genesis = pInfoConfig protocolInfo
+  where
+    protocolInfo :: ProtocolInfo IO (ShelleyBlock c)
+    protocolInfo = protocolInfoShelley
+      genesis
+      NeutralNonce -- TODO
+      2000
+      (SL.ProtVer 0 0)
+      Nothing
+
+countOutputs :: Shelley.Crypto c => SL.Tx c -> Int
+countOutputs tx  = length $ SL._outputs $ SL._body tx
+
+{-------------------------------------------------------------------------------
+  CardanoBlock instance
+-------------------------------------------------------------------------------}
+
+instance TPraosCrypto c => HasAnalysis (CardanoBlock c) where
+  countTxOutputs blk = case blk of
+    Cardano.BlockByron b    -> countTxOutputs b
+    Cardano.BlockShelley sh -> countTxOutputs sh
+  blockHeaderSize blk = case blk of
+    Cardano.BlockByron b    -> blockHeaderSize b
+    Cardano.BlockShelley sh -> blockHeaderSize sh
+  blockTxSizes blk = case blk of
+    Cardano.BlockByron b    -> blockTxSizes b
+    Cardano.BlockShelley sh -> blockTxSizes sh
+  mkTopLevelConfig [byronGenesis, shelleyGenesis] onMainNet = do
+    byronConfig <- openGenesisByron byronGenesis onMainNet
+    shelleyConfig <- either (error . show) return =<< Aeson.eitherDecodeFileStrict' shelleyGenesis
+    return $ mkCardanoTopLevelConfig byronConfig shelleyConfig
+  mkTopLevelConfig ls _onMainNet =
+    error $
+      "Two genesis files are needed for Cardano. Given " ++ show ls
+
+mkCardanoTopLevelConfig :: forall c. TPraosCrypto c
+                        => Genesis.Config
+                        -> ShelleyGenesis c
+                        -> TopLevelConfig (CardanoBlock c)
+mkCardanoTopLevelConfig byronConfig shelleyConfig = pInfoConfig protocolInfo
+  where
+    protocolInfo :: ProtocolInfo IO (CardanoBlock c)
+    protocolInfo = protocolInfoCardano
+      byronConfig
+      Nothing
+      (Update.ProtocolVersion 1 0 0)
+      (Update.SoftwareVersion (Update.ApplicationName "db-analyse") 2)
+      Nothing
+      shelleyConfig
+      NeutralNonce -- TODO
```

Comment:
I guess adding this default value here won't create any issues

Reply:
This is fine for now, but when they start using the hash of the genesis JSON file, we'll have to update this (take it as a parameter?)